scripts/binary/binary-cleanup.js

```diff
@@ -114,13 +113,53 @@ const getDependencyPathsToKeep = async (buildAppDir) => {
   return [...Object.keys(esbuildResult.metafile.inputs), ...entryPoints]
 }
 
-const cleanup = async (buildAppDir) => {
+const createServerEntryPointBundle = async (buildAppDir) => {
+  const unixBuildAppDir = buildAppDir.split(path.sep).join(path.posix.sep)
+  const entryPoints = [path.join(unixBuildAppDir, 'packages/server/index.js')]
+  // Build the binary entry point ignoring anything that happens in start-cypress since that will be in the v8 snapshot
+  const esbuildResult = await esbuild.build({
+    entryPoints,
+    bundle: true,
+    outdir: workingDir,
+    platform: 'node',
+    metafile: true,
+    absWorkingDir: unixBuildAppDir,
+    external: [
+      './transpile-ts',
+      './start-cypress',
+    ],
+  })
+
+  console.log(`copying server entry point bundle from ${path.join(workingDir, 'index.js')} to ${path.join(buildAppDir, 'packages', 'server', 'index.js')}`)
+
+  await fs.copy(path.join(workingDir, 'index.js'), path.join(buildAppDir, 'packages', 'server', 'index.js'))
+
+  console.log(`compiling server entry point bundle to ${path.join(buildAppDir, 'packages', 'server', 'index.jsc')}`)
+
+  // Use bytenode to compile the entry point bundle. This will save time on the v8 compile step and ensure the integrity of the entry point
+  await bytenode.compileFile({
+    filename: path.join(buildAppDir, 'packages', 'server', 'index.js'),
+    output: path.join(buildAppDir, 'packages', 'server', 'index.jsc'),
+    electron: true,
+  })
+
+  return [...Object.keys(esbuildResult.metafile.inputs)].map((input) => `./${input}`)
```

Comment:
Should this be a hard-coded separator?

Reply:
Yes, the way these files are read in assumes "require" syntax not platform specific syntax.